### ISSUE TITLE
Increased ARN checking for check_ignore_list?

### ISSUE THREAD

**Issue by blakestoddard** [Netflix/security_monkey#663]
Hi!

Is there a particular reason that ARNs are ran through `check_ignore_list` on very limited basis when the watchers run? It seems like the ARN checks are limited to very few resources when they could be applied to a lot of other resources (such as `ec2instances`), which would make the Ignore List infinitely more useful as I could ignore stuff based on account.

If there is a reason, do you have any suggestions for other methods that I could try so that I could ignore particular resources from certain accounts that don't currently support having their ARN checked via check_ignore_list?

**Comment by monkeysecurity** [comment]
Hey @blakestoddard,

Each watcher decides what to send to `check_ignore_list`.  Most send the item name.  Managed Policies send both the name and the arn: https://github.com/Netflix/security_monkey/blob/develop/security_monkey/watchers/iam/managed_policy.py#L65

Are you requesting that the other watchers send both name and arn, instead of just name?

**Comment by blakestoddard** [comment]
Yep, that's exactly what I'm asking for. It seems like there are quite a number of other watchers that could send ARNs through, too, just like managed policies do.

**Comment by monkeysecurity** [comment]
Maybe modify `check_ignore_list` to take a list instead of a str:
```
    def check_ignore_list(self, names=None):
        """
        See if the given item has a name flagging it to be ignored by security_monkey.
        """
        if not names:
            return False
        if type(names) is not list:
            names = list(names)
        for name in names:
            for result in self.ignore_list:
                # Empty prefix comes back as None instead of an empty string ...
                prefix = result.prefix or ""
                if name.lower().startswith(prefix.lower()):
                    app.logger.warn("Ignoring {}/{} because of IGNORELIST prefix {}".format(self.index, name, result.prefix))
                    return True

        return False
```

And then in the watchers we can collapse this:
```
                if self.check_ignore_list(policy.policy_name):
                    continue

                if self.check_ignore_list(policy.arn):
                    continue
```

Into this:

```
                if self.check_ignore_list([policy.arn, policy.policy_name]):
                    continue
```

**Comment by monkeysecurity** [comment]
I wonder if we should also consider replacing prefixes with regular expressions.... Or allowing the user to choose which each one should be...

**Comment by blakestoddard** [comment]
The pass-in-an-array-and-iterate-over-it approach seems like a good path to take here.

As for adding regex support, it could be nice? I personally don't have a use for it and would much rather see per-account Ignore Lists (or the ability to specify that a particular ignore case only applies to one account and not universally), but having ARN ignore support in more places would take care of that use case.
